### PR TITLE
Merge string-interpolation docs into scala3 book

### PR DESCRIPTION
**Pull request by rockpunk**
This is my first stab at tackling #2813.  There are many changed files mainly due to renumbering the metadata, however the overview of the change is as follows:

1. Create a new `string-interpolation` page in scala 3 book based on Josh's excellent `_overviews/core/string-interpolation.md`. This is after `first-look-at-types` and before `control-structures.md`
2. redirect `_overviews/core/string-interpolation.md`  to `overviews/scala3-book/string-interpolation.md`
3. Moved the string interpolation stuff out of `first-look-at-types.md`
4. Moved the multi-line string info in `first-look-at-types.md` to its own `Strings` section which describes multiline in more detail and links to new `string-interpolation.md`
5. Modified the custom interpolator example to give a trivial example so readers could cut/copy working code
6. `s/John/James/g` just to be consistent

### CODE REVIEW

File: _overviews/scala3-book/string-interpolation.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+quotes, and prefixing any variable names with a `$` symbol.
+
+### Other interpolators
+
+The `s` that you place before the string is just one possible interpolator that Scala
+provides.
+
+Scala provides three string interpolation methods out of the box:  `s`, `f` and `raw`.
+Further, a string interpolator is a just special method so it is possible to define your
+own. For instance, some database libraries define the very powerful `sql` interpolator
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:38]:
```suggestion
own. For instance, some database libraries define a `sql` interpolator
```

File: _overviews/scala3-book/string-interpolation.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+{% endtab %}
+{% endtabs %}
+
+The raw interpolator is useful when you want to avoid having expressions like `\n` turn into a return character.
+
+In addition to the three default string interpolators, users can define their own.
+
+## Advanced Usage
+
+The literal `s"Hi $name"` is parsed by scala as a _processed_ string literal.
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:225]:
```suggestion
The literal `s"Hi $name"` is parsed by Scala as a _processed_ string literal.
```

File: _overviews/scala3-book/string-interpolation.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+it transforms it into a method call (`id`) on an instance of [StringContext](https://www.scala-lang.org/api/current/scala/StringContext.html).
+This method can also be available on implicit scope.
+To define our own string interpolation, we need to create an implicit class (Scala 2) or an `extension` method (Scala 3) that adds a new method to `StringContext`.
+
+As a trivial example, let's assume we have a simple `Point` class and want to create a custom interpolator that turns `p"a,b"` into a `Point` object.
+
+{% tabs custom-interpolator-1 %}
+{% tab 'Scala 2 and 3' for=custom-interpolator-1 %}
+```scala
+case class Point(x:Double, y:Double)
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:251]:
```suggestion
case class Point(x: Double, y: Double)
```

File: _overviews/scala3-book/string-interpolation.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+```scala
+extension (sc: StringContext)
+  def p(args: Any*): Point = ???
+```
+{% endtab %}
+
+{% endtabs %}
+
+Once this extension is in scope and the Scala compiler encounters `p"some string"`, it
+will process `some $string` to turn it into String tokens and expression arguments.
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:284]:
```suggestion
will process `some string` to turn it into String tokens and expression arguments.
```

File: _overviews/scala3-book/string-interpolation.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+For example, `p"1, $var"` would turn into:
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:285]: `var` cannot be an identifier.

```suggestion
For example, `p"1, $variable"` would turn into:
```

File: _overviews/scala3-book/string-interpolation.md [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+
+{% tabs extension-desugaring class=tabs-scala-version %}
+
+{% tab 'Scala 2' for=extension-desugaring %}
+```scala
+new StringContext("1, ", "").p(var)
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:291]:
```suggestion
new StringContext("1, ", "").p(variable)
```

File: _overviews/scala3-book/string-interpolation.md [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+```
+
+The implicit class is then used to rewrite it to the following:
+
+```scala
+new PointHelper(new StringContext("1, ", "")).p(var)
```
Comment (julienrf) [inline _overviews/scala3-book/string-interpolation.md:297]:
```suggestion
new PointHelper(new StringContext("1, ", "")).p(variable)
```

**Review by julienrf — COMMENTED** [review]
Thank you for your contribution! Please see my review below.

File: _overviews/scala3-book/string-interpolation.md (hunk 5 above)
Comment (rockpunk) [reply _overviews/scala3-book/string-interpolation.md:285]: doh! wasn't thinking

**Review by julienrf — APPROVED** [review]
Thank you very much for your contribution! Do you mind fixing the Git conflicts so that we can merge the PR?

**Comment by bishabosha** [comment]
removing the link from guides and overviews could probably come in a new PR where we make more sweeping changes to the site's navigation

**Comment by bishabosha** [comment]
thanks @rockpunk for all the work!

**Comment by julienrf** [comment]
@bishabosha I thought the entry in Guides & Overviews would be removed when we removed the file `string-interpolation.md`. What should we do instead?

**Pull request merged**